assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,593 @@
+
+#### Ex. 48: Print the _first_ 15 lines of `sam/DPCh_plate1_A05_S5.sam` to _stdout_
+
+(base) [zhimei@colostate.edu@login11 sam]$ cat DPCh_plate1_A05_S5.sam | head -n 15
+@HD     VN:1.3  SO:coordinate
+@SQ     SN:NC_037097.1  LN:96198142
+@SQ     SN:NC_037098.1  LN:57406636
+@SQ     SN:NC_037099.1  LN:81079776
+@SQ     SN:NC_037100.1  LN:74299616
+@SQ     SN:NC_037101.1  LN:93271540
+@SQ     SN:NC_037102.1  LN:77009353
+@SQ     SN:NC_037103.1  LN:86953105
+@SQ     SN:NC_037104.1  LN:73297397
+@SQ     SN:NC_037105.1  LN:90872356
+@SQ     SN:NC_037106.1  LN:60898746
+@SQ     SN:NC_037107.1  LN:48052637
+@SQ     SN:NC_037108.1  LN:77127610
+@SQ     SN:NC_037109.1  LN:75288599
+@SQ     SN:NC_037110.1  LN:55471129
+
+#### Ex. 49: Catenate the file `count_files.sh` to _stdout_
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ cat count_files.sh
+# This looks fun
+ls -lR  * | awk 'NF>7' | wc | awk '{print "Estimated", $1, "files in the current directory"}' 
+
+#### Ex. 50: Note that `count_files.sh` is a short shell script. Try to 
+####         execute it with `./count_files.sh`
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ./count_files.sh
+Estimated 80 files in the current directory
+
+#### Ex. 51: List (long) `count_files.sh` to see what the permissions are,
+####         and then add execute (`x`) permissions to users and groups with
+####         the "add-or-subtract permissions syntax" of `chmod`, then long list it again
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls -l count_files.sh
+-rwxr-xr-x. 1 zhimei@colostate.edu zhimeigrp@colostate.edu 112 Jan 24 10:45 count_files.sh
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ chmod +x count_files.sh
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls -l count_files.sh
+-rwxr-xr-x. 1 zhimei@colostate.edu zhimeigrp@colostate.edu 112 Jan 24 10:45 count_files.sh
+
+
+#### Ex. 52: Use chmod to change permissions of the files in the `silly`
+####         directory to match their names. For example for the first 
+####         one you would use: `chmod 770 FILE-rwxrwx---`. Use the octal
+####         specifiers to `chmod`. Do this for all 5 files (one chmod command for each),
+####         and at the end do `ls -l silly` to verify they are correct.
+
+(base) [zhimei@colostate.edu@login11 silly]$ chmod 777 DIRdrwxrwxr-x/
+(base) [zhimei@colostate.edu@login11 silly]$ ls -l
+total 128
+drwxrwsrwx. 2 zhimei@colostate.edu zhimeigrp@colostate.edu 23 Jan 24 10:45 DIRdrwxrwxr-x
+-rwxrwx---. 1 zhimei@colostate.edu zhimeigrp@colostate.edu  0 Jan 24 10:45 FILE-rw-r-----
+-rwxrwx---. 1 zhimei@colostate.edu zhimeigrp@colostate.edu  0 Jan 24 10:45 FILE-rw-rw-r--
+-rwxrwx---. 1 zhimei@colostate.edu zhimeigrp@colostate.edu  0 Jan 24 10:45 FILE-rwxr-----
+-rwxrwx---. 1 zhimei@colostate.edu zhimeigrp@colostate.edu  0 Jan 24 10:45 FILE-rwxrwxr--
```

Comment:
We talked about this a little bit in class.  Do the octal permissions specifiers make more sense now?